src/SelfService/Infrastructure/Api/CustomObjectResults.cs

```diff
@@ -2,33 +2,22 @@
 
 namespace SelfService.Infrastructure.Api;
 
-public static class CustomObjectResults
+public class CustomObjectResult : ObjectResult
 {
-    [NonAction]
-    public static InternalServerErrorResult InternalServerError(object error) => new(error);
-
-    [NonAction]
-    public static MethodNotAllowedErrorResult MethodNotAllowedError(object error) => new(error);
-}
-
-public class InternalServerErrorResult : ObjectResult
-{
-    private const int DefaultStatusCode = StatusCodes.Status500InternalServerError;
-
-    public InternalServerErrorResult(object error)
+    private CustomObjectResult(int statusCode, object error)
         : base(error)
     {
-        StatusCode = DefaultStatusCode;
+        StatusCode = statusCode;
     }
-}
 
-public class MethodNotAllowedErrorResult : ObjectResult
-{
-    private const int DefaultStatusCode = StatusCodes.Status405MethodNotAllowed;
+    [NonAction]
+    public static CustomObjectResult InternalServerError(object error) =>
+        new(StatusCodes.Status405MethodNotAllowed, error);
```

Comment:
A bit strange that we are returning 405 for an Internal Server Error

Reply:
:P thats a bug. Fixed! Also renamed file to match class name